api/handlers/events_handler.go

```diff
@@ -69,4 +72,59 @@ func (h *Handler) GetAllEvents(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 	}
+}
+
+// Query Event by their eventID
+func (h *Handler) GetEventByEventID(w http.ResponseWriter, r *http.Request) {
+	fmt.Println(" - Called GetEventByEventID")
+	eventID := chi.URLParam(r, "eventID")
+	if eventID == "" {
+		fmt.Println("eventID = ", eventID)
+		// If eventID is empty, return an error response
+		http.Error(w, "eventID parameter is required", http.StatusBadRequest)
+		return
+	}
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*3)
+	defer cancel()
+
+	query := "SELECT * FROM events WHERE eventID = $1"
+	row, err := h.db.QueryContext(ctx, query, eventID)
+	fmt.Println("row= ", row)
+	if err != nil {
+		fmt.Println(err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	defer row.Close()
+	
+	// Scan rows to create Brother instance
+	var event models.Event
+	for row.Next() {
+		event, err = createEventFromRow(row)
+		if err != nil {
+			log.Fatal("Error!", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+	}
+
+	if event.EventID == 0 {
```

Comment:
Why do we need this codeblock for?

Reply:
This code block is for if the eventID passed doesn't exist in the table then it will return the http error in the block. It is implemented in this way because through testing on Postman, I saw that when inputting non-existing ID, it produces default values such as 0 for eventID and other fields.